fearless/Modules/WalletMainContainer/WalletMainContainerInteractor.swift

```diff
@@ -120,8 +120,15 @@ extension WalletMainContainerInteractor: WalletMainContainerInteractorInput {
 
 extension WalletMainContainerInteractor: EventVisitorProtocol {
     func processWalletNameChanged(event: WalletNameChanged) {
-        selectedMetaAccount = event.wallet
-        output?.didReceiveAccount(selectedMetaAccount)
+        if wallet.identifier == event.wallet.identifier {
+            wallet = event.wallet
+            output?.didReceiveAccount(wallet)
+        }
+    }
+
+    func processMetaAccountChanged(event: MetaAccountModelChangedEvent) {
+        wallet = event.account
```

Comment:
mb rename event.account to event.wallet to unify? It's already wallet in previous event.